src/nix/profile.cc

```diff
@@ -41,6 +41,12 @@ struct ProfileElement
     bool active = true;
     int priority = defaultPriority;
 
+    bool hasStorePath(const StorePath & basePath) const {
+        return any_of(storePaths.begin(), storePaths.end(), [&](const StorePath & storePath) {
+            return storePath == basePath;
+        });
+    }
```

Comment:
I like the combinators! But try https://en.cppreference.com/w/cpp/container/set/contains or (if our C++20 is not new enough) https://en.cppreference.com/w/cpp/container/set/find

Reply:
Then you don't need this method

Reply:
I missed this method in my search for a right candidate. That makes it quite a bit cleaner :+1: